src/components/AddMediaModalWrapper/AddMediaModalWrapper.js

```diff
@@ -26,6 +27,7 @@ function AddMediaModalWrapper({
     // allowing user to attach the selected/uploaded files to the document.
     const handleOnClick = () => {
       if (tabHasSelectedItems) updateSavedMedia(selectedMedia)
+      setSelectedMedia([])
```

Comment:
I think there is a bigger problem here than how we handle switching tabs. It appears that the selector is not distinguishing between the full list of selected items (the value of the form control) and the list of items that are currently selected in the modal. Looks like this affects the EntryArrayField as well.

I'm going to need to think about this a bit to figure out the best place to adjust this.